src/MonoTorrent.Client/MonoTorrent/Torrent.cs

```diff
@@ -426,7 +426,7 @@ void ProcessInfo (BEncodedDictionary dictionary, ref PieceHashesV1? hashesV1)
                     case ("files"):
                         // This is the list of files using the v1 torrent format.
                         // Only load if we have not processed filesv2
-                        if (Files.Count == 0)
+                        // if (Files.Count == 0) // TODO JMIK: merge v1 hashes info with v2 ?? 
```

Comment:
I'm lazy, so I use https://chocobo1.github.io/bencode_online for viewing the content of .torrent files when I want to review exactly what's in there.

I used that site to view the hybrid torrent linked in the libtorrent blog ( https://blog.libtorrent.org/2020/09/bittorrent-v2/ ) to see what impact choosing the V1 or V2 files would have.

The difference is:
- Padding files only exist in V1 metadata
- `pieces root` only exists with V2 metadata.

In order to support hybrid torrents, there is one way I can think of to avoid merging data, and it also provides a little bit of extra safety for incorrectly generated torrents.

For a standards compliant hybrid torrent we know padding files are always exactly: `TorrentManager.PieceLength - file.Length % TorrentManager.PieceLength` bytes long. (They always extend the file so it comes the remaining space in that piece).

As such, when we call [LoadTorrentFilesV2](https://github.com/alanmcgovern/monotorrent/blob/002708a4220eb4c2d2b6d950c35d5bdae7712184/src/MonoTorrent.Client/MonoTorrent/Torrent.cs#L435) we could pass in an additional `isHybrid` boolean (i.e. `hasV1Data  && hasV2Data` ) and set the correct padding value for the files we parse if the torrent is a hybrid v1/v2 torrent.

Later, when we parse the V1 metadata we can verify that the *actual* padding length parsed from the v1 'files' key matches the value we generated when parsing the v2 files from `files tree`. If it doesn't match then it's not a standards compliant file and we can abort parsing.

I think that solves the problem, while also adding a sanity check that the padding files were created correctly :)

Reply:
Added that in my latest commit, the validation works for the hybrid test torrent. The hybrid torrent loading test fails with MonoTorrent.TorrentException : the 'piece layers' dictionary did not contain an entry for the file 'tbl-tint.mpg'.
I think I spotted a rounding bug in v2 EndPieceIndex calculation..